llvm/include/llvm/Analysis/EHUtils.h

```diff
@@ -16,6 +16,7 @@ namespace llvm {
 /// Compute a list of blocks that are only reachable via EH paths.
 template <typename FunctionT, typename BlockT>
 static void computeEHOnlyBlocks(FunctionT &F, DenseSet<BlockT *> &EHBlocks) {
+  assert(EHBlocks.empty() && "Output set should be empty");
```

Comment:
I understand that you want to make sure we don't do `findInvokeNormalDests` first, only to have it cleared here. But this assertion is not compatible with the `EHBlocks.clear()` later in this function. If we clear it explicit, we are not expecting it to be empty, vice versa.

Reply:
Good point, then going to remove the `EHBlocks.clear()`, I think it should be user's responsibility to choose the right input, the function should avoid this side effect.